src/lib/utils/column_ids_after_pruning.cpp

```diff
@@ -0,0 +1,33 @@
+#include "column_ids_after_pruning.hpp"
+
+#include <cstdlib>
+#include <optional>
+#include <vector>
+
+namespace opossum {
+
+const std::vector<std::optional<ColumnID>> column_ids_after_pruning(const size_t& original_table_column_count,
```

Comment:
This should not be an issue, but I believe that we normally do not pass plain data types as `size_t` by reference. There is no overhead in passing it by value and it could only confuse the reader.